### PR TITLE
BUGFIX make pagination work with Traversable

### PR DESCRIPTION
**Pull request by ru3fu5z**
Related #1112

Warning: array_slice() expects parameter 1 to be array, object given in /tmp/Flow/Development/SubContextddev/Cache/Code/Flow_Object_Classes/Neos_ContentRepository_ViewHelpers_Widget_Controller_PaginateController.php line 124

### CODE REVIEW

File: Neos.ContentRepository/Classes/ViewHelpers/Widget/Controller/PaginateController.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ public function indexAction($currentPage = 1)
         }
         $offset = ($this->currentPage > 1) ? (integer)($itemsPerPage * ($this->currentPage - 1)) : null;
         if ($this->parentNode === null) {
-            $nodes = array_slice($this->nodes, $offset, $itemsPerPage);
+            $nodes = (array) array_slice($this->nodes, $offset, $itemsPerPage);
```
Comment (mhsdesign) [inline Neos.ContentRepository/Classes/ViewHelpers/Widget/Controller/PaginateController.php:124]: i would much rather change it here in: initializeAction()

`$this->nodes = $this->widgetConfiguration['nodes'];`

something like `if $this->widgetConfiguration['nodes'] instaceof TraversableNodes` ...

**Review by mhsdesign — CHANGES_REQUESTED** [review]

File: Neos.ContentRepository/Classes/ViewHelpers/Widget/Controller/PaginateController.php (hunk 1 above)
Comment (mhsdesign) [reply Neos.ContentRepository/Classes/ViewHelpers/Widget/Controller/PaginateController.php:129]: also by reading your provided excepfion - your fix wont work i assume. youd need to "transform" the first param to an array but via getIterator()->getArrayCopy() or toArray()

Comment (ru3fu5z) [reply Neos.ContentRepository/Classes/ViewHelpers/Widget/Controller/PaginateController.php:129]: Thank for your help. I updated the branch with your recommented changes

**Comment by jonnitto** [comment]
@mhsdesign Do you think 8.2 is the correct branch?

**Comment by mhsdesign** [comment]
Nope should be 7.3 - im not a fluid gui - but id like to see a test maybe ^^

**Review by jonnitto — CHANGES_REQUESTED** [review]
@ru3fu5z, thank you for your PR. I think the best is, to close this PR and create a new one, based on the correct branch.

**Pull request closed without merging**